File: 27. Remove Element/removeElement.cpp
#include <iostream>
#include <vector>

using namespace std;

int removeElement(vector<int> &nums, int val) {
  for (int i = 0; i < nums.size();) {
    if (nums[i] == val)
      nums.erase(nums.begin() + i);
    else
      i++;
  }
  return nums.size();
}

int main() {
  vector<int> nums{3, 2, 2, 3};
  int val = 3;
  cout << removeElement(nums, val);
}

/*
[3,2,2,3], 3 = 2;
[0,1,2,2,3,0,4,2], 2 = 5
*/

File: 27. Remove Element/removeElement.js
var removeElement = function (nums, val) {
  // nearly identical to 26. Remove Duplicates from Sorted Array
	for (let i = nums.length - 1; i >= 0; i--) { 
		if (nums[i] === val) {
			nums.splice(i, 1);
		}
	}
	return nums.length;
};

const arr = [3, 2, 2, 3];
const num = 2;
console.log(removeElement(arr, num));

/*
[3,2,2,3], 3 = 2;
[0,1,2,2,3,0,4,2], 2 = 5
*/
